src/rdb_protocol/terms/sort.cc

```diff
@@ -83,9 +83,9 @@ class orderby_term_t : public op_term_t {
         /* Add a sorting to the table if we're doing indexed sorting. */
         } else if (index.has()) {
             rcheck(tbl_slice.has(), base_exc_t::LOGIC,
-                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE.");
+                   "Indexed order_by can only be performed on a TABLE or TABLE_SLICE, not on a `NAME_OF_TYPE_ORDER_BY_WAS_CALLED_ON`.  Make sure ORDER_BY comes before any transformations such as MAP or FILTER.");
```

Comment:
I don't feel too strongly either as long as it's consistent.  I remember the convention being uppercase, so my guess would be that a lot of the older code uses uppercase and most of the lowercase examples are newer code.  (The lower-case `order_by` you linked was added by Nighelles earlier this year, which is consistent with that.)

If you think the lower-case is nicer I'd be happy to move to that.